src/lib.rs

```diff
@@ -212,6 +212,62 @@ impl<I, P> FixedPoint<I, P> {
     }
 }
 
+/// Avoids the use of [`std::io::Cursor`] in `no_std` context
+struct Cursor<T> {
+    buffer: T,
+    position: usize,
+}
+impl<T> Cursor<T> {
+    fn new(buffer: T) -> Self {
+        Self {
+            buffer,
+            position: 0,
+        }
+    }
+    fn position(&self) -> usize {
+        self.position
+    }
+}
+impl<T: AsMut<[u8]>> fmt::Write for Cursor<T> {
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        let raw_s = s.as_bytes();
+        let pos = self.position;
+
+        let buf = &mut self.buffer.as_mut()[pos..pos + raw_s.len()];
+
+        if raw_s.len() > buf.len() {
+            return Err(fmt::Error);
+        }
+
+        buf.copy_from_slice(raw_s);
+        self.position += raw_s.len();
+
+        Ok(())
+    }
+}
+
+#[cfg(feature = "std")]
+fn f64_significant_digits(value: f64) -> usize {
+    value.abs().log10().ceil() as usize
+}
+
+#[cfg(not(feature = "std"))]
+fn f64_significant_digits(value: f64) -> usize {
+    let mut abs_value = if value.is_sign_negative() {
+        -value
+    } else {
+        value
+    };
+
+    let mut significant = 0;
+    while abs_value > 1f64 {
```

Comment:
Although it doesn't matter in the whole context, it seems there should be `abs_value >= 1f64` for `f64_significant_digits(1.0)` case.

Add test (in new created module) that this function matches `value.abs().log10().ceil()`.

Reply:
added tests. I think previous implementation was faulty as well